cli/azd/pkg/tools/azcli/account.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &AzCliSubscriptionInfo{
+		Id:       subscriptionResponse.Value.SubscriptionID,
+		Name:     subscriptionResponse.Value.DisplayName,
+		TenantId: subscriptionResponse.Value.TenantID,
+	}, nil
+}
+
+func (cli *azCli) GetSubscriptionTenant(ctx context.Context, subscriptionId string) (string, error) {
+	subscription, err := cli.GetAccount(ctx, subscriptionId)
+	if err != nil {
+		return "", err
+	}
+
+	return subscription.TenantId, nil
+}
+
+func (cli *azCli) ListAccountLocations(ctx context.Context, subscriptionId string) ([]AzCliLocation, error) {
+	client, err := cli.createSubscriptionsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	locations := []AzCliLocation{}
+	pager := client.NewListLocationsPager(subscriptionId, nil)
+
+	var rawResponse *http.Response
+	ctx = runtime.WithCaptureResponse(ctx, &rawResponse)
+
+	for pager.More() {
+		_, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed getting next page of locations: %w", err)
+		}
+
+		// Using custom response processing unto Go SDK support required properties
+		locationResponse, err := readRawResponse[CustomListLocationsResponse](rawResponse)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, location := range locationResponse.Value {
+			// Ignore non-physical locations
+			if location.Metadata.RegionType != "Physical" {
+				continue
+			}
+
+			locations = append(locations, AzCliLocation{
+				Name:                location.Name,
+				DisplayName:         location.DisplayName,
+				RegionalDisplayName: location.RegionalDisplayName,
+			})
+		}
+	}
+
+	sort.Slice(locations, func(i, j int) bool {
+		return locations[i].RegionalDisplayName < locations[j].RegionalDisplayName
+	})
+
+	return locations, nil
+}
+
+func (cli *azCli) createSubscriptionsClient(ctx context.Context) (*armsubscription.SubscriptionsClient, error) {
+	cred, err := identity.GetCredentials(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Uses latest api version of subscriptions api to get additional properties
+	options := cli.createArmClientOptions(ctx, convert.RefOf("2020-01-01"))
+	client, err := armsubscription.NewSubscriptionsClient(cred, options)
+	if err != nil {
+		return nil, fmt.Errorf("creating Subscriptions client: %w", err)
+	}
+
+	return client, nil
+}
+
+// Temporary custom response until Go SDK support all of the specified properties in the response
+type CustomListLocationsResponse struct {
+	Value []*CustomLocation `json:"value"`
+}
+
+// Temporary custom response until Go SDK support all of the specified properties in the response
+type CustomLocation struct {
+	ID                  string                 `json:"id"`
+	Name                string                 `json:"name"`
+	DisplayName         string                 `json:"displayName"`
+	RegionalDisplayName string                 `json:"regionalDisplayName"` // Missing from Go SDK
```

Comment:
I recall that you may have opened issues for these missing values.   Could we link to the issues in the comments here?

Reply:
added.